presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixSplitManager.java

```diff
@@ -71,23 +71,22 @@ public PhoenixSplitManager(PhoenixClient phoenixClient)
     }
 
     @Override
-    public ConnectorSplitSource getSplits(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorTableLayoutHandle layout, SplitSchedulingStrategy splitSchedulingStrategy)
+    public ConnectorSplitSource getSplits(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorTableHandle table, SplitSchedulingStrategy splitSchedulingStrategy)
     {
-        PhoenixTableLayoutHandle layoutHandle = (PhoenixTableLayoutHandle) layout;
-        JdbcTableHandle handle = layoutHandle.getTable();
+        JdbcTableHandle tableHandle = (JdbcTableHandle) table;
         try (PhoenixConnection connection = phoenixClient.getConnection(JdbcIdentity.from(session))) {
-            List<JdbcColumnHandle> columns = layoutHandle.getDesiredColumns()
+            List<JdbcColumnHandle> columns = tableHandle.getColumns()
```

Comment:
I think `PhoenixMetadata#applyProjection()` should also be implemented, otherwise this will fetch all the columns when only a subset is being queried for.

Reply:
Since we have extended it from `JdbcMetadata` it would be applied